packages/node-cli/src/commands/object/update/UpdateAction.js

```diff
@@ -47,26 +53,90 @@ module.exports = class UpdateAction extends BaseAction {
 			if (params.hasOwnProperty(ANSWERS_NAMES.OVERWRITE_OBJECTS) && !params[ANSWERS_NAMES.OVERWRITE_OBJECTS]) {
 				throw NodeTranslationService.getMessage(MESSAGES.CANCEL_UPDATE);
 			}
-			const sdkParams = CommandUtils.extractCommandOptions(params, this._commandMetadata);
+			let updateCustomRecordsWithInstancesResult = [];
+			if (params.hasOwnProperty(COMMAND_OPTIONS.INCLUDE_INSTANCES) && params[COMMAND_OPTIONS.INCLUDE_INSTANCES]) {
+				const customRecordScriptIds = params[COMMAND_OPTIONS.SCRIPT_ID]
+					.split(' ')
+					.filter((scriptId) => this._isCustomRecordOrSegment(scriptId));
+				params[COMMAND_OPTIONS.SCRIPT_ID] = params[COMMAND_OPTIONS.SCRIPT_ID]
+					.split(' ')
+					.filter((scriptId) => !this._isCustomRecordOrSegment(scriptId))
+					.join(' ');
+				updateCustomRecordsWithInstancesResult = await this._updateCustomRecordWithInstances(params, customRecordScriptIds);
 
-			const executionContextForUpdate = SdkExecutionContext.Builder.forCommand(this._commandMetadata.sdkCommand)
-				.integration()
-				.addParams(sdkParams)
-				.build();
+				delete params[COMMAND_OPTIONS.INCLUDE_INSTANCES];
+			}
 
-			const operationResult = await executeWithSpinner({
-				action: this._sdkExecutor.execute(executionContextForUpdate),
-				message: NodeTranslationService.getMessage(MESSAGES.UPDATING_OBJECTS),
-			});
+			if (params[COMMAND_OPTIONS.SCRIPT_ID] === '') {
+				// If there are no more objects to update (all were custom records) we already return with only previous messages
+				return ActionResult.Builder.withData(updateCustomRecordsWithInstancesResult).build();
+			}
 
-			return operationResult.status === SdkOperationResultUtils.STATUS.SUCCESS
-				? ActionResult.Builder.withData(operationResult.data)
-					.withResultMessage(operationResult.resultMessage)
-					.withCommandParameters(sdkParams)
-					.build()
-				: ActionResult.Builder.withErrors(operationResult.errorMessages).withCommandParameters(sdkParams).build();
+			const sdkParams = CommandUtils.extractCommandOptions(params, this._commandMetadata);
```

Comment:
This line could be inside `this._updateObjects` and just pass the `params` as parameters.

Reply:
Hey David, I have put this out, because sdkParams in needed below. I don't want to call two times to CommandUtils.extractCommandOptions

This is because you asked to get out the following code from  _updateObjects.

So either we go to the previous implementation, or we stay with sdkParams here

Reply:
Oh, I didn't realize about it. Then I think we can leave them here.